### PR TITLE
Add Discord mod log

### PR DESCRIPTION
**Pull request by 12pm**
Logs usage of `ban`, `kick` and `muteid` to a discord channel with a webhook
image

### CODE REVIEW

**Comment by Learath2** [comment]
If you are making this discord specific you could link curl into the server again and just do the http request from there. However, if you are making it more generic, I'd just remove "Discord" from the code and use something like "ModLogServer"

**Comment by heinrich5991** [comment]
Dislike that we're making up a new protocol here. Can we just link in curl again and to a HTTP JSON request instead?

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -1340,6 +1340,26 @@ void CServer::ProcessClientPacket(CNetChunk *pPacket)
 					Console()->SetAccessLevel(IConsole::ACCESS_LEVEL_ADMIN);
 					m_RconClientID = IServer::RCON_CID_SERV;
 					m_RconAuthLevel = AUTHED_ADMIN;
+
+					if(g_Config.m_SvDiscordModLogPort != -1 && (str_startswith(pCmd, "ban") || str_startswith(pCmd, "kick") || str_startswith(pCmd, "muteid")))
```
Comment (heinrich5991) [inline src/engine/server/server.cpp]: This does not catch all bans/kicks/etc. You need to hook into the commands instead. Currently, you could prefix the command, use multiple commands in one line, etc. to circumvent this.

**Comment by 12pm** [comment]
then i need to do a lot string escaping in c++

File: src/engine/server/server.cpp (hunk 1 above)
Comment (Learath2) [reply src/engine/server/server.cpp]: Yep, you could also use ConChains for these to make it pretty. e.g. register chains for all these commands using a `LogEvent` static.

**Comment by heinrich5991** [comment]
Instead of implementing a custom protocol via UDP, this should be doing something like HTTP requests. I'm closing this for now.

**Pull request closed without merging**